dbt/adapters/spark/impl.py

```diff
@@ -390,13 +418,27 @@ def _get_one_catalog(
 
         database = information_schema.database
         schema = list(schemas)[0]
+        relations = self.list_relations(database, schema)
+        return self._get_relation_metadata_at_column_level(relations)
 
+    def _get_relation_metadata_at_column_level(self, relations: List[BaseRelation]) -> agate.Table:
```

Comment:
This was the second half of `_get_one_catalog` above. It is broken out as it's own method now so it can be reused by `_get_one_catalog_by_relations` below.